program/src/processor/mod.rs

```diff
@@ -0,0 +1,152 @@
+use borsh::BorshDeserialize;
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, msg, program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+use crate::instruction::{
+    accounts::{
+        DeactivateStakeAccounts, DistributeRewardsAccounts, HarvestHolderRewardsAccounts,
+        HarvestStakeRewardsAccounts, InactivateStakeAccounts, InitializeConfigAccounts,
+        InitializeStakeAccounts, SetAuthorityAccounts, SlashAccounts, StakeTokensAccounts,
+        UpdateConfigAccounts, WithdrawInactiveStakeAccounts,
+    },
+    Instruction,
+};
+
+mod deactivate_stake;
+mod distribute_rewards;
+mod harvest_holder_rewards;
+mod harvest_stake_rewards;
+mod inactivate_stake;
+mod initialize_config;
+mod initialize_stake;
+mod set_authority;
+mod slash;
+mod stake_tokens;
+mod update_config;
+mod withdraw_inactive_stake;
+
+#[inline(always)]
+pub fn process_instruction<'a>(
+    program_id: &Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    // TODO: should use Pod types for instruction data?
```

Comment:
I think so, yeah, or just implement the packing/unpacking by hand. The instruction data is very small (at most a couple of `u64` values), so we really don't need Borsh.

Reply:
Done!